types/yup/index.d.ts

```diff
@@ -212,18 +212,29 @@ export type TransformFunction<T> = ((
     originalValue: any
 ) => any);
 
-export interface WhenOptionsBuilder<T> {
+export interface WhenOptionsBuilderFunction<T> {
```

Comment:
I think we should call this type `WhenOptionsFunction<T>` because the docs call the second argument `builder` when it's an object.

Reply:
`builder` is the name of the argument when it is an object or function. I was trying to stick with the existing naming scheme, but add more specificity.

Reply:
Yeah, okay. I'll concede. 😄